synapse/rest/client/pusher.py

```diff
@@ -57,7 +57,9 @@ async def on_GET(self, request: SynapseRequest) -> Tuple[int, JsonDict]:
         for pusher in pusher_dicts:
             if self._msc3881_enabled:
                 pusher["org.matrix.msc3881.enabled"] = pusher["enabled"]
+                pusher["org.matrix.msc3881.device_id"] = pusher["device_id"]
```

Comment:
Can this hold a `null` value? If so, are clients expected to be able to understand this?

Reply:
Yes, as per the MSC, this field is nullable: https://github.com/matrix-org/matrix-spec-proposals/blob/kerry/remote-push-toggle/proposals/3881-remote-push-notification-toggling.md#explicitly-linking-device-and-pusher